index.ts

```diff
@@ -44,14 +44,14 @@ export class DataLoaderInterceptor implements NestInterceptor {
     const ctx: any = graphqlExecutionContext.getContext();
 
     if (ctx[NEST_LOADER_CONTEXT_KEY] === undefined) {
-      ctx[NEST_LOADER_CONTEXT_KEY] = (type: string): NestDataLoader<any, any> => {
+      ctx[NEST_LOADER_CONTEXT_KEY] = async (type: string) => {
         if (ctx[type] === undefined) {
           try {
-            ctx[type] = this.moduleRef
-              .get<NestDataLoader<any, any>>(type, { strict: false })
+            ctx[type] = (await this.moduleRef
+              .resolve<NestDataLoader<any, any>>(type, undefined, { strict: false }))
```

Comment:
I'm not very familiar with the resolve method, but by passing in a value of undefined, won't it always generate a new instance of the requested data loader class? I think we would want them to be the same per request: https://docs.nestjs.com/fundamentals/module-ref#retrieving-instances

Reply:
Thank you for pointing this out! I'm able to get the request out of the graphqlExecutionContext and get a contextId. Please let me know what would think!